### PR TITLE
[feature](alter colocate group) Support alter colocate group replica allocation

### PR DESCRIPTION
**Pull request by yujun777**
## Proposed changes

Support alter colocate group's replica allocation.
1. SQL command:   `ALTER COLOCATE GROUP [database.]group SET ("replication_allocation" = "...")`. 

2. Colocate group belongs to a database, but if its name starts with `__global__`, then it's a global group, and does not belong to any database;	

3. After alter group schema's replica allocations,  also modify all the group's related tables's  propertities `default.replication_allocation` and `dynamic.replication_allocation`,  also modify all the table's existing partitions's replica allocations;

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by yujun777** [comment]
run buildall

File: docs/en/docs/sql-manual/sql-reference/Data-Definition-Statements/Alter/ALTER-COLOCATE-GROUP.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+SET (
+    property_list
+);
+```
+
+NOTE:
+
+1. `full_group_name` is the full name of the colocation group, which can be divided into two cases:
+ - If the group is global, that is, its name starts with `__global__`, then `full_group_name` is equal to `group_name`;
+ - If the group is not global, that is, its name does not start with `__global__`, then it belongs to a certain Database, `full_group_name` is equal to `dbId` + `_` + `group_name`
```
Comment (dataroaring) [inline docs/en/docs/sql-manual/sql-reference/Data-Definition-Statements/Alter/ALTER-COLOCATE-GROUP.md:52]: dbname is better than dbid here.

File: docs/en/docs/sql-manual/sql-reference/Data-Definition-Statements/Alter/ALTER-COLOCATE-GROUP.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+ALTER COLOCATE GROUP "full_group_name"
+SET (
+    property_list
+);
+```
+
+NOTE:
+
+1. `full_group_name` is the full name of the colocation group, which can be divided into two cases:
+ - If the group is global, that is, its name starts with `__global__`, then `full_group_name` is equal to `group_name`;
```
Comment (dataroaring) [inline docs/en/docs/sql-manual/sql-reference/Data-Definition-Statements/Alter/ALTER-COLOCATE-GROUP.md:51]: Users should not care about __global__.

File: docs/en/docs/sql-manual/sql-reference/Data-Definition-Statements/Alter/ALTER-COLOCATE-GROUP.md (hunk 1 above)
Comment (dataroaring) [reply docs/en/docs/sql-manual/sql-reference/Data-Definition-Statements/Alter/ALTER-COLOCATE-GROUP.md:52]: There is __global__ in the example below.

**Review by dataroaring — CHANGES_REQUESTED** [review]

File: docs/en/docs/sql-manual/sql-reference/Data-Definition-Statements/Alter/ALTER-COLOCATE-GROUP.md (hunk 2 above)
Comment (yujun777) [reply docs/en/docs/sql-manual/sql-reference/Data-Definition-Statements/Alter/ALTER-COLOCATE-GROUP.md]: done.  users can see group name from  `show proc "/colocation_group"`

**Comment by yujun777** [comment]
run buildall

**Comment by yujun777** [comment]
run buildall

**Comment by yujun777** [comment]
run buildall

**Comment by dataroaring** [comment]
run buildall

**Comment by yujun777** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Review by kaijchen — APPROVED** [review]

**Pull request merged**